sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -166,6 +177,14 @@ class SparkSessionExtensions {
     runtimeOptimizerRules += builder
   }
 
+  /**
+   * Inject a rule that can override the query stage optimizer phase of adaptive query
+   * execution.
+   */
+  def injectQueryStageOptimizerRule(builder: QueryStagePrepRuleBuilder): Unit = {
```

Comment:
Oh, this looks like a type of `QueryStageOptimizerRuleBuilder`. We need a builder of type `QueryStageOptimizerRuleBuilder` instead of `QueryStagePrepRuleBuilder` type, right?